deps/obs-scripting/CMakeLists.txt

```diff
@@ -1,216 +1,180 @@
-cmake_minimum_required(VERSION 2.8.12)
+option(ENABLE_SCRIPTING_LUA "Enable Lua scripting support" ON)
+option(ENABLE_SCRIPTING_PYTHON "Enable Python scripting support" ON)
 
 if(NOT ENABLE_SCRIPTING)
-	message(STATUS "Scripting plugin disabled")
-	return()
+  message(STATUS "OBS: Scripting plugin disabled")
+  return()
 endif()
 
 project(obs-scripting)
 
-if(POLICY CMP0068)
-	# RPATH settings on macOS do not affect install_name.
-	cmake_policy(SET CMP0068 NEW)
+if(ENABLE_SCRIPTING_LUA)
+  add_subdirectory(obslua)
+  find_package(Luajit)
+
+  if(NOT TARGET Luajit::Luajit)
+    message(FATAL_ERROR "OBS: Luajit not found - support disabled")
+    return()
+  else()
+    message(STATUS "OBS: Luajit found - support enabled")
+  endif()
+else()
+  message(STATUS "OBS: Luajit support disabled")
+endif()
+
+if(ENABLE_SCRIPTING_PYTHON)
+  add_subdirectory(obspython)
+  if(OS_WINDOWS)
+    find_package(PythonWindows)
+  else()
+    find_package(Python COMPONENTS Interpreter Development)
+  endif()
+
+  if(NOT TARGET Python::Python)
+    message(FATAL_ERROR "OBS: Python not found - support disabled")
+    return()
+  else()
+    message(STATUS "OBS: Python ${Python_VERSION} found - support enabled")
+  endif()
+else()
+  message(STATUS "OBS: Python support disabled")
 endif()
 
-if(MSVC)
-	set(obs-scripting_PLATFORM_DEPS
-		w32-pthreads)
+if(NOT TARGET Luajit::Luajit AND NOT TARGET Python::Python)
+  message(
+    WARNING "OBS: No supported scripting libraries found; plugin disabled")
+  return()
 endif()
 
-if(APPLE)
-	set(obs-scripting_PLATFORM_DEPS
-		objc)
+if(OS_MACOS)
+  find_package(SWIG 4 REQUIRED)
+elseif(OS_POSIX)
+  find_package(SWIG 3 REQUIRED)
+elseif(OS_WINDOWS)
+  find_package(SwigWindows 3 REQUIRED)
```

Comment:
Minor nitpick:  in this file and in deps/obs-scripting/obslua/CMakeLists.txt and deps/obs-scripting/obspython/CMakeLists.txt, the order of the if/elseif block for finding SWIG isn't consistent.

Reply:
Fixed.